packages/htm/src/render.ts

```diff
@@ -1,65 +1,38 @@
 import baseHtm from 'htm';
+import {isRemoteComponent} from '@remote-ui/core';
 import type {RemoteRoot, RemoteComponent} from '@remote-ui/core';
 
-interface Element {
-  type: string;
-  props: Record<string, any>;
-  children: (string | Element)[];
-}
+export function createRender(root: RemoteRoot<any, any>) {
+  return baseHtm.bind((type, props, ...children) => {
+    const normalizedChildren: (string | RemoteComponent<any, any>)[] = [];
 
-export const htm = baseHtm.bind<Element>((type, props, ...children) => {
-  if (typeof type !== 'string') {
-    throw new Error(`You can only create components with string names.`);
-  }
+    for (const child of children) {
+      if (child == null || child === false) continue;
```

Comment:
Also, is there a reason `reduce()` isn't used here in place of the `for` loop?  It's accumulating into `normalizedChildren`

Reply:
We use `== null` for this feature pretty often, IMO it's only gotten more acceptable in JS to do so since that's basically the same check that `?.` and `??` use. I could have used a reduce here, but I wanted to keep memory consumption down as much as possible here — which means only creating a single array for holding the children — and I find reduce looks awkward when you just mutate the collector array, instead of creating a new array on each call.